assembl/views/api2/discussion.py

```diff
@@ -740,12 +745,21 @@ def get_visitors(request):
     use_first = asbool(request.GET.get("first", False))
     attribute = "first_visit" if use_first else "last_visit"
     visitors = []
+    from assembl.models import ProfileField
     for st in discussion.agent_status_in_discussion:
         if not getattr(st, attribute, None):
             continue
+        profile_fields = db.query(ProfileField).filter(ProfileField.discussion_id == discussion.id).filter(
+            ProfileField.agent_profile_id == st.agent_profile.id).all()
         data = {"time": getattr(st, attribute),
                 "name": (st.agent_profile.name or '').encode("utf-8"),
                 "email": (st.agent_profile.get_preferred_email() or '').encode("utf-8")}
+
+        for configurable_field in configurable_fields:
+            for profile_field in profile_fields:
+                if profile_field.configurable_field_id == configurable_field.id:
+                    data.update({configurable_field.title.entries[0].value.encode("utf-8"): profile_field.value_data['value'].encode("utf-8")})
```

Comment:
shouldn't you take the title for the chosen language?

Reply:
yes, you need to use `.best_lang(user_prefs)` instead of `.entries[0]` here.